app/src/main/java/org/mozilla/tv/firefox/IntentValidator.kt

```diff
@@ -57,10 +60,18 @@ object IntentValidator {
 
         when (intent.action) {
             Intent.ACTION_MAIN -> {
-                val dialParams = intent.extras?.getString(DIAL_PARAMS_KEY) ?: return null
-                if (dialParams.isNotEmpty()) {
+                val extras = intent.extras ?: return null
+
+                if (extras.keySet().contains(DIAL_PARAMS_KEY)) {
+                    val param = extras.getString(DIAL_PARAMS_KEY)
                     TelemetryIntegration.INSTANCE.youtubeCastEvent()
-                    return ValidatedIntentData(url = "https://www.youtube.com/tv?$dialParams", source = Session.Source.ACTION_VIEW)
+                    return ValidatedIntentData(url = "https://www.youtube.com/tv?$param", source = Session.Source.ACTION_VIEW)
+                }
+                // The fetch delay can be changed for testing purposes through an intent with custom flags
```

Comment:
nit: maybe we should pull this into a helper function. A general practice I follow is that if I'm writing a comment for code, I'll consider pulling it into a separate method. The reasons are:
- if the code is complicated enough to warrant a comment, it should probably be abstracted/isolated in a function
- the helper function name can often stand-in for a comment
- when comments are added, the code being commented is sometimes lower-level than the code around it and it's hard to read code that is at different levels of abstraction